aws/resource_aws_lex_intent_test.go

```diff
@@ -0,0 +1,379 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lexmodelbuildingservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// Intents can accept a custom SlotType but it must be removed from the Intent before the SlotType can be deleted.
+// This means we cannot reference the SlotType in the Intent with interpolation because Terraform will try to delete
+// the SlotType first which will fail. So we do not have a test for custom slot types.
+
+func TestAccAwsLexIntent_basic(t *testing.T) {
```

Comment:
This test should be split up. Typically, we prefer that the `_basic` test is a simple test, possibly with an import and/or simple update. The breakdown for slot types is a good guide.

It would definitely be good to have a test that validates version publishing.